### PR TITLE
Support for dynamically detect new catalogs

### LINKED ISSUE #2445: Support for dynamically detect new catalogs or deleted catalogs
I need presto to be able to add catalogs dynamically without restarting presto server. 

So generally, if I add a new catalog file to presto, without a "launcher restart" command the server can automatically detect it. And once a catalog is removed(suppose it is not in use), the server will also detect it.

Our system is like this: We have a table in mysql storing all the data servers(IP , port, type, username and password). And when a new data source is added to the system, a new catalog is added to the server who has presto server on it. We cannot afford presto server to restart to get it refreshed. Instead, we want it to detect that change automatically.

### PR DESCRIPTION
**Pull request by yuananf**
Fix #2445

### CODE REVIEW

**Comment by cberner** [comment]
It looks like the tests are failing. Can you take a look at Travis?

**Comment by yuananf** [comment]
@cberner Fixed

File: presto-main/src/main/java/com/facebook/presto/server/AnnouncementUtils.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static io.airlift.discovery.client.ServiceAnnouncement.ServiceAnnouncementBuilder;
+import static io.airlift.discovery.client.ServiceAnnouncement.serviceAnnouncement;
+
+public final class AnnouncementUtils
+{
+    private AnnouncementUtils() {}
+
+    public static void updateDatasources(Announcer announcer, Metadata metadata, ServerConfig serverConfig, NodeSchedulerConfig schedulerConfig)
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/server/AnnouncementUtils.java]: Can you pull this refactoring into a separate commit?

Comment (yuananf) [inline presto-main/src/main/java/com/facebook/presto/server/AnnouncementUtils.java]: sure

Comment (yuananf) [inline presto-main/src/main/java/com/facebook/presto/server/AnnouncementUtils.java]: @electrum done.

**Comment by martint** [comment]
Thanks for your submission. Unfortunately, this is not the approach we'd like to pursue. There are a number of issues that the mechanism for loading/unloading catalogs needs to address. See my comment on the related issue: https://github.com/prestodb/presto/issues/2445#issuecomment-223983285

**Comment by yuananf** [comment]
@martint ok, thanks for explaining.

**Comment by SonuSingh200190** [comment]
do we have any update on Dynamically Detect New/Deleted Catalogs?
is it possible to manage catalogs only on coordinate node instead of worker node?
